tests/test_units/test_reader_tsurf.py

```diff
@@ -0,0 +1,217 @@
+"""Test the tsurf readers module"""
+
+from pathlib import Path
+
+import numpy as np
+
+from fmu.dataio._readers import tsurf_file_reader as reader
+
+
+def _create_hardcoded_file(filepath: Path):
+    """Create a hardcoded TSurf file for testing purposes."""
+
+    with open(filepath, "w") as file:
+        file.write("""GOCAD TSurf 1
+HEADER {
+name: Fault F1
+}
+GOCAD_ORIGINAL_COORDINATE_SYSTEM
+NAME Default
+AXIS_NAME "X" "Y" "Z"
+AXIS_UNIT "m" "m" "m"
+ZPOSITIVE Depth
+END_ORIGINAL_COORDINATE_SYSTEM
+TFACE
+VRTX 1 0.1 0.2 0.3 CNXYZ
+VRTX 2 1.1 1.2 1.3 CNXYZ
+VRTX 3 2.1 2.2 2.3 CNXYZ
+VRTX 4 3.1 3.2 3.3 CNXYZ
+TRGL 1 2 3
+TRGL 1 2 4
+END
+""")
+
+
+def _create_dummy_TSurfFileData():
```

Comment:
If a function is used several in a test, make it a pytest fixture. This will ensure that setup and teardowns in tests is done. See e.g. `test_export_rms_volumetrics.py`